Added the possibility to select a project.

swc-pdi opened the pull request:
- I have made some changes on the climate-tagger.php file
- I don't have changed the tests, expect the adaption of the API URL
- I added a new change-log notice in the readme.txt, file but I don't changed the version number

moserk committed:
Added the possibility to select a project.

daniell commented:
Could you update the version number in the readme.txt file and add some tests for your functionality for getting the thesauri.

moserk committed:
Changed the short description for the new feature.

moserk committed:
Adapted methods after ClimateTagger API changes

swc-pdi commented:
I have made some changes on methods, after the ClimateTagger API was changed.
You can merge this Pull Request now. Thanks.

daniell commented:
The code for getting the thesauri is still untested. Once tests for it are added, I'll merge the pull request.

moserk committed:
Added an unit test for project selecting

swc-pdi commented:
Hi daniell, I added the unit test for the selecting of a project.
Please, can you merge the pull request. Thank you.

swc-pdi commented:
Hello Aptivate,
please can you check my adaptations and the unit test integration and can you please update the plugin on WordPress. Thank you very much.

martinburchell commented:
Thanks @swc-pdi  for your patience - we are busy at the moment with paid work but I'll aim to get this done soon

martinburchell commented on climate-tagger.php:
```diff
@@ -12,6 +12,9 @@
 }
 
+// Documentation: http://api.climatetagger.net/documentation/
+define('CLIMATE_TAGGER_API_URL', 'http://api.climatetagger.net');
```
WordPress coding standards require spaces before and after parentheses

swc-pdi replied:
Done

martinburchell commented on tests/mock-remote-get.php:
```diff
@@ -0,0 +1,19 @@
+
+	$_CLIMATE_TAGGER_MOCK_URL = $url;
+
+  $response['response']['code'] = 200;
```
Use tabs to indent lines

swc-pdi replied:
Done

martinburchell commented on tests/test-plugin.php:
```diff
@@ -173,6 +174,26 @@ public function test_word_limit_retrieved_from_options() {
+			$project,
+			$this->equalTo( 'Climate Change Adaptation' )
+		);
+  }
```
Use tabs to indent (line 195)

swc-pdi replied:
Done

martinburchell reviewed (changes requested):
Thanks for submitting this pull request. You've added some useful functionality to the plugin.

As a general point, it makes it harder to approve pull requests that have a lot of changes. There are a few distinct changes here, each of which should have had their own pull request eg:
- Ability to select project
- Change of climate tagger API URL
- Link to settings from plugin page
- Fixing PHP warnings
- Code and HTML formatting changes

anyway if you could address the comments here, I'll approve the pull request and release a new version of the plugin

moserk committed:
Implemented requested changes on the pull request.

swc-pdi commented on climate-tagger.php:
```diff
@@ -2,7 +2,7 @@
   Plugin Name: Climate Tagger
   Description: Recommends tags in a tag cloud based on Climate Tagger API.
-  Version: 1.0.2
+  Version: 1.0.4
```
You are right. This was a mistake.

swc-pdi commented on readme.txt:
```diff
@@ -3,7 +3,7 @@ Contributors: Aptivate
 Requires at least: 3.7
 Tested up to: 4.2.2
-Stable tag: 1.0.2
+Stable tag: 1.0.4
```
You are right. This was a mistake.

moserk committed:
Implemented requested changes on the pull request for the tests also.

swc-pdi commented on climate-tagger.php:
```diff
@@ -177,19 +213,63 @@ function print_tag_cloud( $tags_rec ) {
+					$key = 'default';
+				}
+				$projects[ $key ] = array(
+					'label' => $project,
```
Also here I have simplified the code.

swc-pdi commented:
Thank you for the great review. I adapted and simplified some code parts.
Further I have changed the methods for the ClimateTagger class to static functions, because I have got many "Strict Standards" (E_STRICT) warnings. Calling the functions using the double colon syntax must be a static function, because they were interpreted as static (since the PHP version 5.3.0).

Thank you for your help,
Kurt

martinburchell commented on climate-tagger.php:
```diff
@@ -12,117 +12,149 @@
+				'limit'      => '20',
 				'post_types' => 'post',
-			) );
+				'project'    => 'default',
```
You can simply remove `'project' => 'default'` now because if the plugin is upgraded and the project setting is not changed by the admin, you will end up sending `'default'` as the tagger to the API, which will result in an internal server error being returned.

swc-pdi replied:
You're absolutely right. Sorry for this error.

martinburchell reviewed (commented):
Thanks for making the changes and the further quality improvements. I think we're nearly there!

moserk committed:
Last adaptations for the pull request

swc-pdi commented:
Thanks martinburchell for all.

martinburchell commented:
Thanks @swc-pdi
